packages/mdc-textfield/_mixins.scss

```diff
@@ -201,78 +201,150 @@
   @include mdc-notched-outline-shape-radius($radius, $rtl-reflexive);
 }
 
+///
+/// Customizes the color of the text entered into an enabled text field.
+/// @param {Color} $color - The desired input text color.
+///
 @mixin mdc-text-field-ink-color($color) {
   &:not(.mdc-text-field--disabled) {
     @include mdc-text-field-ink-color_($color);
   }
 }
 
-@mixin mdc-text-field-fill-color($color) {
-  &:not(.mdc-text-field--disabled) {
-    @include mdc-text-field-fill-color_($color);
+///
+/// Customizes the color of the entered text in a disabled text field.
+/// @param {Color} $color - The desired input text color.
+///
+@mixin mdc-text-field-disabled-ink-color($color) {
+  &.mdc-text-field--disabled {
+    @include mdc-text-field-ink-color_($color);
   }
 }
 
-@mixin mdc-text-field-textarea-stroke-color($color) {
+///
+/// Customizes the background color of the text field or textarea when enabled.
+/// @param {Color} $color - The desired background color.
+///
+@mixin mdc-text-field-fill-color($color) {
   &:not(.mdc-text-field--disabled) {
-    @include mdc-text-field-textarea-stroke-color_($color);
+    @include mdc-text-field-fill-color_($color);
   }
 }
 
-@mixin mdc-text-field-textarea-fill-color($color) {
-  &:not(.mdc-text-field--disabled) {
-    @include mdc-text-field-textarea-fill-color_($color);
+///
+/// Customizes the background color of the text field or textarea when disabled.
+/// @param {Color} $color - The desired label text color.
+///
+@mixin mdc-text-field-disabled-fill-color($color) {
+  &.mdc-text-field--disabled {
+    @include mdc-text-field-fill-color_($color);
   }
 }
 
-@mixin mdc-text-field-fullwidth-bottom-line-color($color) {
+///
+/// Customizes the text field bottom line color except the outlined and textarea variants.
+/// @param {Color} $color - The desired bottom line color.
+///
+@mixin mdc-text-field-bottom-line-color($color) {
   &:not(.mdc-text-field--disabled) {
-    @include mdc-text-field-fullwidth-bottom-line-color_($color);
+    @include mdc-text-field-bottom-line-color_($color);
   }
 }
 
-@mixin mdc-text-field-bottom-line-color($color) {
-  &:not(.mdc-text-field--disabled):not(.mdc-text-field--outlined):not(.mdc-text-field--textarea) {
+///
+/// Customizes the disabled text field bottom line color except the outlined and textarea variants.
+/// @param {Color} $color - The desired bottom line color.
+///
+@mixin mdc-text-field-disabled-bottom-line-color($color) {
+  &.mdc-text-field--disabled {
     @include mdc-text-field-bottom-line-color_($color);
   }
 }
 
+///
+/// Customizes the hover text field bottom line color except the outlined and textarea variants.
```

Comment:
Instead of indicating that this mixin does not work for certain text field variants, I think we can say that it only works for filled text field variant.

Reply:
Same for above.